test/function.localstack.test.ts

```diff
@@ -0,0 +1,332 @@
+      };
+
+      return {
+        outputs: {
+          function1: create("func6", "c").resource.functionName,
+          function2: create("func7", "d").resource.functionName,
+        },
+      };
+    },
+    async (context) => {
+      await testFunction(context.function1, {}, "c");
+      await testFunction(context.function2, {}, "d");
+    }
+  );
+
+  testResource(
+    "Call Lambda with object",
+    (parent) => {
+      const create = () => {
+        const obj = { val: 1 };
+        return new Function(parent, "function", async () => obj.val);
+      };
+
+      return { outputs: { function: create().resource.functionName } };
+    },
+    async (context) => {
+      await testFunction(context.function, {}, 1);
+    }
+  );
+
+  testResource(
+    "Call Lambda with math",
+    (parent) => {
+      const func = new Function(parent, "function", async () => {
+        const v1 = 1 + 2; // 3
+        const v2 = v1 * 3; // 9
+        return v2 - 4; // 5
+      });
+
+      return {
+        outputs: { function: func.resource.functionName },
+      };
+    },
+    async (context) => {
+      await testFunction(context.function, {}, 5);
+    }
+  );
+
+  testResource(
+    "Call Lambda payload",
+    (parent) => {
+      const func = new Function(
+        parent,
+        "function",
+        async (event: { val: string }) => {
+          return `value: ${event.val}`;
+        }
+      );
+
+      return { outputs: { function: func.resource.functionName } };
+    },
+    async (context) => {
+      await testFunction(context.function, { val: "hi" }, "value: hi");
+    }
+  );
+
+  testResource(
+    "Call Lambda throw error",
+    (parent) => {
+      const func = new Function(parent, "function", async () => {
+        throw Error("AHHHHHHHHH");
+      });
+
+      return {
+        outputs: {
+          function: func.resource.functionName,
+        },
+      };
+    },
+    async (context) => {
+      await testFunction(
+        context.function,
+        {},
+        { errorMessage: "AHHHHHHHHH", errorType: "Error" }
+      );
+    }
+  );
+
+  testResource(
+    "Call Lambda put event to bus",
+    (parent) => {
+      const bus = new EventBus(parent, "bus");
+      const busbus = new aws_events.EventBus(parent, "busbus");
+      const func = new Function(parent, "function", async () => {
+        return `${bus.eventBusArn} ${busbus.eventBusArn}`;
+      });
+
+      return {
+        outputs: {
+          bus: bus.eventBusArn,
+          busbus: busbus.eventBusArn,
+          function: func.resource.functionName,
+        },
+      };
+    },
+    async (context) => {
+      await testFunction(
+        context.function,
+        {},
+        `${context.bus} ${context.busbus}`
+      );
+    }
+  );
+
+  testResource(
+    "Call Lambda AWS SDK put event to bus with reference",
+    (parent) => {
+      const bus = new EventBus<any>(parent, "bus");
+
+      // Necessary to keep the bundle small and stop the test from failing.
```

Comment:
How significantly does this effect the bundle size?

Reply:
3mb vs 100kb (including the serialized coverage code).

Reply:
Can we fix this now that we have control over the closure serializer?